pkg/didcomm/protocol/exchange/exchange.go

```diff
@@ -86,29 +91,39 @@ func GenerateInviteWithKeyAndEndpoint(invite *Invitation) (string, error) {
 	return encodedExchangeInvitation(invite)
 }
 
+// Protocol for exchange protocol
+type Protocol struct {
+	prov transport.OutboundTransport
+}
+
+// New instanstiated new exchange client
+func New(prov provider) *Protocol {
```

Comment:
May be we need a public interface here, if in case someone wants to use 'exchange.Protocol' in standalone fashion (without using aries.New(...) )?  In future if 'exchange.Provider' interface grows? @troyronda @Baha-sk

Reply:
This was changed based on [comments](https://github.com/hyperledger/aries-framework-go/pull/88#discussion_r313590180).

Reply:
Add some notes/comments if we are continuing with private interface.

Reply:
[Updated](https://github.com/hyperledger/aries-framework-go/compare/a31807a201229cd5ab417413af54914995b98b87..a26c159c381cbbb380ef97f616c8f9b9bddd476a#diff-4d3ca323f9b237cfa9940c12efa9b615R100).